man/fi_efa.7.md

```diff
@@ -205,6 +205,12 @@ These OFI runtime parameters apply only to the RDM endpoint.
   [`ptrace protection`](https://wiki.ubuntu.com/SecurityTeam/Roadmap/KernelHardening#ptrace_Protection)
   is turned on. You can turn it off to enable shm transfer.
 
+  FI_EFA_ENABLE_SHM_TRANSFER is related to the FI_OPT_SHARED_MEMORY_PERMITTED endpoint option.
+  If FI_EFA_ENABLE_SHM_TRANSFER is set to true, the FI_OPT_SHARED_MEMORY_PERMITTED endpoint
+  option overrides FI_EFA_ENABLE_SHM_TRANSFER. If FI_EFA_ENABLE_SHM_TRANSFER is set to false,
+  but the FI_OPT_SHARED_MEMORY_PERMITTED is set to true, the FI_OPT_SHARED_MEMORY_PERMITTED
```

Comment:
Why do we want to fail instead of always trusting the env var?

Reply:
We shouldn't ignore the conflicting configuration. We have the same handling for use_device_rdma

Reply:
Because if the env var is set to false, we don't create SHM resources in the `fi_domain` call. I don't think we should go back and re-create those resources.

I can update the man page to clarify that `FI_EFA_ENABLE_SHM_TRANSFER` is parsed in the `fi_domain` call